src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  actions.forEach((action) => {
+    if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
```

Comment:
The line `currentState = { ...currentState };` is redundant since you are making a copy of the currentState without any changes. You can remove this line as the currentState will be copied again at the beginning of the next action if necessary.